_components/code-component.md

```diff
@@ -0,0 +1,221 @@
+you could if you have extended it yourself). In our Dashboard start building your
+integration and include the Code component as well. You will see a picture similar
+to the one below:
+
+![Code component](/assets/img/components/code-component/code-component01.png "Code component")
+
+However, don't let the _simple look_ fool you - it has a full-fledged interface
+with many very useful features like the ones you would expect from your favourite
+desktop developing tool:
+
+*   **Syntax highlighting** - a standard for any online real-time coding interface
+*   **Code auto-completion** - again a standard that you got used to from your desktop tool
+*   Support for number of **[variables and libraries](#available-variables-and-libraries)** within the context of the execution
+*   Support for [ECMAScript 2015](https://en.wikipedia.org/wiki/ECMAScript) JavaScript standard
+*   Run and troubleshoot within the designer interface.
+
+### Requirements
+
+#### Environment variables
+
+Code component does not need environment variables to function. Environment Variables
+can be used to tune or extend the functionality like available RAM memory for the
+docker containers to use.
+
+## Credentials
+
+Code component does not require credentials to function.
+
+## Triggers
+
+### Executes custom code
+
+This trigger executes a custom code inputted on the interface.
+
+## Actions
+
+### Executes custom code
+
+This action executes a custom code inputted on the interface.
+
+## Additional info
+
+### Available Variables and Libraries
+
+Here are the available variables and libraries that can be used within the context
+of execution. The most up-to-date list can always be found in be used within the
+context of execution. Below is the list for your reference:
+
+*   `console`: - more on [Node.js console](https://nodejs.org/dist/latest-v5.x/docs/api/console.html),
+*   `process`: - [Current Node.js process](https://nodejs.org/dist/latest-v5.x/docs/api/process.html),
+*   `require`: - [Module require](https://nodejs.org/dist/latest-v5.x/docs/api/modules.html),
+*   `setTimeout`: - more on [setTimeout](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `clearTimeout`: - more on [clearTimeout](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `setInterval`: - more on [setInterval](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `clearInterval`: - more on [clearInterval](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `msg`: - Incoming message containing the payload from the previous step,
+*   `exports`: {},
+*   `messages`: - [Utility for convenient message creation](https://github.com/elasticio/elasticio-node/blob/master/lib/messages.js),
+*   `request`: - [Http Client](https://github.com/id0x3d/co-request),
+*   `wait`: - wait,
+*   `emitter`: user to emit messages and errors
+
+
+## Known limitations
+
+Only the code written in `Node.js` is supported at this moment.
+
+## Code component usage Examples
```

Comment:
Most of these methods are super-dated, but then so is the component.  Perhaps limit it to the callback section and scrap the other two sections?